discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -229,14 +235,19 @@ def set_host_roles(self, num_masters: int = None, num_workers: int = None, reque
         assigned_roles.extend(
             self._get_matching_hosts(host_type=consts.NodeRoles.WORKER, count=requested_roles["worker"])
         )
-
-        self.api_client.update_hosts(cluster_id=self.id, hosts_with_roles=assigned_roles)
+        for role in assigned_roles:
+            self.api_client.update_host(
+                infra_env_id=infra_env['id'],
+                host_id=role['id'],
+                host_role=role['role']
+            )
 
         return assigned_roles
 
     def set_specific_host_role(self, host, role):
```

Comment:
Same here .. I think that this method should be on infra_env and not in cluster

Reply:
calling the api through the infra_env. wdyt?